server/src/main/java/org/jboss/as/server/deployment/module/descriptor/JBossDeploymentStructureParser10.java

```diff
@@ -522,8 +522,22 @@ private static void parseResourceRoot(final DeploymentUnit deploymentUnit, final
                             Closeable closable = null;
                             if(overlay != null) {
                                 overlay.remountAsZip(false);
-                            } else if(child.isFile()) {
-                                closable = VFS.mountZip(child, child, TempFileProviderService.provider());
+                            } else {
+                                if(!child.exists()) {
+                                    // the reference does not exist
+                                    throw ServerLogger.ROOT_LOGGER.illegalResoureRootReference(child.getName());
+                                } else if(child.isFile()) {
+                                    try {
+                                        // check if the resource-root is within the deployment or not
+                                       if(child.getPathNameRelativeTo(deploymentRootFile) != null) {
```

Comment:
This check won't work. The JavaDoc for `VirtualFile.getPathNameRelativeTo()` states:
> Get the path name relative to a parent virtual file. If the given virtual file is not a parent of this virtual file, then an IllegalArgumentException is thrown.

Note this is true for each parser, but I'll just comment on this one.